torchmultimodal/models/gpt.py

```diff
@@ -475,6 +491,14 @@ def forward(
         return_attn_weights: bool = False,
         return_hidden_states: bool = False,
     ) -> TransformerDecoderOutput:
+        if attn_mask is not None and attn_mask.dim() == 2:
```

Comment:
Yeah I now better understand some of your comments around `get_extended_attention_mask`. Agree that it would be nice to find a way to support this case as well (since the end result is the same), but not necessary for this PR.